### PR TITLE
Explore more courses section

### PR DESCRIPTION
**Pull request by pawelkmpt**
Let me know if anything can be done better

https://app.clickup.com/t/865bfrhzh

### CODE REVIEW

**Comment by pawelkmpt** [comment]
> This component may not be a right fit for us. After taking a look at the documentation(https://vaadin.com/docs/latest/components/virtual-list) the component expects the data and template to be separated and written in JS.

So we can't just inject custom HTML like I did in Storybook example? It works ;)
What do you suggest instead?

**Comment by anoblet** [comment]
The benefit of the component is that if you have a long list of items(grid with thousands of rows with action buttons), each row will not be added to the DOM until it is needed, and removed when it is not needed. This saves time on first load, and reduces memory usage.

Since we are already providing the content in the DOM, the virtual list isn't having any effect. You can verify this by inspecting the virtual list element and scroll up and down. Nothing is added or removed.

Keeping it simple is way more performant in this case since a few dozen related courses takes less time to load, and less memory used than registering and using the virtual list component.

I wouldn't mind a second opinion on this, though I'm fairly certain this is the case :)

**Comment by lkraav** [comment]
> Since we are already providing the content in the DOM, the virtual list isn't having any effect. You can verify this by inspecting the virtual list element and scroll up and down. Nothing is added or removed.

AFAIK @anoblet is exactly right with his diagnosis.

**Comment by pawelkmpt** [comment]
All clear. Thanks @anoblet for explaining!

I'd be great if Vaadin docs had such information in place...

**Comment by pawelkmpt** [comment]
@anoblet please take a look now

File: packages/storybook/cxl-lumo-styles/courses-list.stories.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          </li>
+        `
+      )}
+    </ul>
+`;
+
+export const CoursesList = Template.bind({});
+
+CoursesList.story = {
+  name: 'Virtual List',
```
Comment (anoblet) [inline packages/storybook/cxl-lumo-styles/courses-list.stories.js:63]:
```suggestion
  name: 'Courses List',
```

**Review by anoblet — APPROVED** [review]
I added one suggestion to update the story name, otherwise LGTM.

**Review by anoblet — APPROVED** [review]

File: packages/cxl-ui/package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "@vaadin/text-field": "^23.3.7",
     "@vaadin/tooltip": "^23.3.7",
     "@vaadin/vaadin-themable-mixin": "^23.3.7",
+    "@vaadin/virtual-list": "^24.1.0",
```
[the remark this answers is not in the document]
Comment (lkraav) [reply packages/cxl-ui/package.json]: Upgrading to v24 is highly likely to have side effects, based on issues I've seen opened. I'd recommend keeping everything at v23 for now.

**Pull request merged**